include/circt/Dialect/OM/OMTypes.td

```diff
@@ -42,4 +42,11 @@ def SymbolRefType : TypeDef<OMDialect, "SymbolRef", []> {
   let mnemonic = "sym_ref";
 }
 
+
+def StringType : TypeDef<OMDialect, "String", []> {
+  let summary = "A type that represents a reference to a string.";
```

Comment:
Is this actually a reference to a string? That makes me think of llvm::StringRef. Maybe we can just say "represents a string"?